### PR TITLE
Reddev 591 report changes

### PR DESCRIPTION
**Pull request by wsams**
If a report is deleted the body of the card presents a message saying the report does not exist.

screen shot 2019-01-02 at 10 34 56

### CODE REVIEW

File: js/components/ReportSummary.vue [hunk 1]
```diff
@@ -28,13 +28,20 @@
...
+        <div v-if="model.reportExists">
+          <ul class="summary-metadata lead list-unstyled mb-0">
+            <li>Total Count: {{ model.totalRecords }}</li>
+            <li v-if="isItemized" class="mt-0">Grouped By: {{ model.bucketByLabel }}</li>
+          </ul>
+          <ul v-if="isItemized" class="list-unstyled mt-3 mb-0" data-description="itemized-counts">
+            <li v-for="summary in sortedSummaryCounts" :key="summary.label">{{ summary.count }} - {{ summary.label }}</li>
+          </ul>
+        </div>
+        <div v-else>
```
Comment (heathharrelson) [inline js/components/ReportSummary.vue]: From looking at the screenshot, I think the div containing the alert could use some top margin. It looks a bit cramped.

I might also rephrase this message to make it clearer. "The report used for this summary no longer exists", or "The report used for this summary was deleted", perhaps?

Comment (lawhead) [reply js/components/ReportSummary.vue]: Agreed

File: js/components/ReportSummaryForm.vue [hunk 2]
```diff
@@ -149,7 +149,7 @@ export default {
 
     // capture the promise to synchronize tests
     this.reportPromise = this.fetchReports();
-    if (model.reportId) {
+    if (model.reportId && model.reportExists) {
```
Comment (heathharrelson) [inline js/components/ReportSummaryForm.vue]: Would this be better as a computed property, like `this.fetchReportFields`?

Comment (heathharrelson) [reply js/components/ReportSummaryForm.vue]: On second thought, `shouldLoadReportFields` might be a better name.

**Pull request merged**